tests/foreman/ui/test_host.py

```diff
@@ -1933,6 +1933,24 @@ def test_all_hosts_bulk_cve_reassign(
             assert row['Lifecycle environment'] == lce2.name
 
 
+@pytest.mark.tier2
+def test_all_hosts_redirect_button(target_sat):
+    """Verify that the New UI button on the old Host page correctly redirects
+    to the All Hosts UI
+
+    :id: 7256f6d0-3ad9-471c-9e3e-bd41cc00a217
+
+    :expectedresults: New UI Button redirects to All Hosts page
+
+    :CaseComponent: Hosts-Content
+
+    :Team: Phoenix-subscriptions
+    """
+    with target_sat.ui_session() as session:
+        url = session.host.new_ui_button()
+        assert "/new/hosts" in url
```

Comment:
I would suggest not to return any value from `new_ui_button()` function, use this function to click on new ui button
Instead of asserting over return value, we can simply check URL endpoint in session object, (tested locally)
```suggestion
        session.host.new_ui_button()
        assert '/new/hosts' in session.browser.url
```

Reply:
I prefer to keep browser type things in the airgun, so that robottelo just calls it like a library, and we don't have robottelo accessing things like that where possible

Reply:
Okay, I would say, let's change the function name to something like `get_new_ui_url()` or something meaningful, as this function returning url.